server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -426,8 +426,11 @@ private void deleteStaleRemotePrimaryTerms() {
         // are older files that are no longer needed and should be cleaned up. In here, we delete all files that are part
         // of older primary term.
         if (olderPrimaryCleaned.trySet(Boolean.TRUE)) {
+            if (readers.isEmpty()) {
+                logger.info("Translog reader list is empty");
+                return;
+            }
             // First we delete all stale primary terms folders from remote store
-            assert readers.isEmpty() == false : shardId + " Expected non-empty readers";
```

Comment:
Was this assert tied to a wrong assumption, i.e. is there a valid scenario where we could be presented with empty readers collection and it would be a valid state to be in?

Reply:
This comment can be resolved. I was able to get the reference for this change via #9458 😃